wyvern/components/models/base_ranking_model.py

```diff
@@ -0,0 +1,52 @@
+# -*- coding: utf-8 -*-
+import asyncio
+from typing import List, Union
+
+from wyvern.components.models.model_component import (
+    MODEL_INPUT,
+    MODEL_OUTPUT,
+    ModelComponent,
+    SingleEntityInput,
+)
+
+
+class BaseRankingModel(ModelComponent[MODEL_INPUT, MODEL_OUTPUT]):
+    async def single_inference(
+        self,
+        input: SingleEntityInput,
+    ) -> Union[float, str, List[float]]:
+        """
+        Single inference for ranking model
+        """
+        raise NotImplementedError
+
+    async def batch_inference(
+        self,
+        input: MODEL_INPUT,
+    ) -> MODEL_OUTPUT:
+        """
+        Batch inference for ranking model
+        """
+        model_score_batches = await asyncio.gather(
+            *[
+                self.single_inference(
+                    SingleEntityInput(entity=entity, request=input.request),
+                )
+                for entity in input.entities
```

Comment:
this way we can't utilize vectorization, do we plan doing that soon?

I think we should have a vectorized batch implementation and we should use that batch function within single inference as well.

Reply:
I thought the feature vector comes with the feature_map but not the MODEL_INPUT.

MODEL_INPUT is the data structure containing all the relative entities.

Reply:
the vectorization part doesn't make sense, ignore that 😄 

but I think it's a more common practice for the single inference to call batch function. so let's do it